### DISCUSSION TITLE
How does Viem know whether or not an account is authorized to send transactions?

### DISCUSSION THREAD

**Discussion by thenoahhein** [wevm/viem#472] [Question]
Hey wagmi team, Phantom dev here. This is 99% i'm sure a bug on our end, but wanted to post this here in case any other wallets end up hitting the same issue.

I'm not 100% sure on the reason for this, I think it's related to us not having support for wallet_requestPermissions enabled.

But the problem is if you are using wagm (viem) you will be unable to send any transactions through the hooks or other transaction APIs

You can reproduce by getting this repo up and running: https://github.com/nheingit/bug-repo-phantom-wagmi

or this site I deployed: https://incandescent-kleicha-a918e4.netlify.app/

And you can connect phantom to the injected connector, and then scroll down to either of the transaction builders and attempt to send a transaction. 

The modal should successfully pop up, and then after approving the transaction, you will get the error: Error: The requested method and/or account has not been authorized by the user. Request Arguments: from: <WALLET_ADDRESS> to:  <TX_TO_VAL> value: <ETH_VALUE> ETH Details: rpcErrorUnauthorizedWrongAccount Version: viem@0.3.18

If you try this with MetaMask it will work.

I only tested on Goerli, but I imagine the network is not the issue here.

**Comment by touka-0** [comment]
Hello, we migrated our app to use wagmi/viem, and then some users reported a similar issue when sending transactions. Is there any progress on fixing or finding a workaround for this?

**Comment by xiao99xiao** [comment]
@nheingit Hi, may I ask what is the status for this bug?(If it is a bug on phantom side)

**Reply by thenoahhein** [reply]
This shouldn't be an issue for you anymore. Be sure to update Phantom

**Comment by Soniachniy** [comment]
I have the same issue with metamask that tried to use sending transaction with phones via scanning desctop QR
the same problem with trustwallet I suppose.
